### PR TITLE
Get dns records v3

### PR DESCRIPTION
**Pull request by dev-0pz**
should be the one that ignores all of the files but otherwise I will need help doing this

### CODE REVIEW

File: reverseforwardrecord/get_dns_record.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+"""
+
+
+def create_file():
+    """
+    makes a txt file for the results to be returned into
+    :return: output.txt
+    """
+    f = open("ignored/output.txt", "w")
+    f.close()
```
Comment (anish-mudaraddi) [inline reverseforwardrecord/get_dns_record.py:24]: you're hardcoding this filepath - you should allow user to enter a filepath via an function parameter 
i.e.
```
from pathlib import Path

def create_file(dirpath):
     """
    makes a txt file for the results to be returned into called output.txt in directory filepath given
    :param: a string representing a directory path
    """
    f = open(Path(filepath / "output.txt"), "w")
    f.close()

File: reverseforwardrecord/get_dns_record.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+convert into list of dictionaries containing them as keys - dns name and ip address
+write a function that takes 2 lists and converts them into a list of dicts
+just have to change read from netbox
+only converts from xlsx
+xlsx file has to have fqdn in column 8 and idrac_ip in column 10
+otherwise have to set it up with manual user input
+"""
+
+
+def create_file():
```
Comment (anish-mudaraddi) [inline reverseforwardrecord/get_dns_record.py:18]: you should add a test for this function - look at mocking open

File: reverseforwardrecord/get_dns_record.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    """
+    create_file()
+    netbox_info = read_from_netbox(netbox_filepath, fqdn_column, idrac_ip_column)
+    parsed_netbox_info = parse_netbox_info(netbox_info, reverse_order)
+    write_output(parsed_netbox_info, output_filepath, reverse_order)
+
+
+if __name__ == "__main__":
+    netbox_filepath = "ignored/test.xlsx"
+    main(netbox_filepath, reverse_order=False)
```
Comment (anish-mudaraddi) [inline reverseforwardrecord/get_dns_record.py:131]: we're going to need to add argparse to this so we can read in command-line arguments - that's the next job. 

Currently the user of this script needs to modify values here to use it properly. Instead we want the user to be able to specify arguments via the command line when calling the script i.e.

`python get_dns_records.py --netbox_filepath "/path/to/file.xlsx" --reverse True` etc

**Review by anish-mudaraddi — CHANGES_REQUESTED** [review]
looks good - but really this isn't complete until you add arg-parse and read from parameters 
https://docs.python.org/3/library/argparse.html

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,7 @@
+*
+
+!.gitignore
+!get_dns_record.py
+!test_get_dns_record.py
+!requirements.txt
+!reverseforwardrecord/.github/workflows/get_dns_records.yaml
```
Comment (khalford) [inline .gitignore:7]:
```suggestion
.idea/
__pycache__/
venv/
.vscode/
```

Comment (khalford) [reply .gitignore:7]: It's better to ignore the files you don't want than to ignore everything except the files you do want.

**Review by khalford — CHANGES_REQUESTED** [review]
You should move `.github/workflows` to the top level directory.

File: .gitignore (hunk 4 above)
Comment (dev-0pz) [reply .gitignore:7]: no because sensitive information file names change around so it is safer to do it this way

**Comment by DavidFair** [comment]
@dev-0pz Can you see if we can get this moving again?

**Comment by khalford** [comment]
Closing due to inactivity. Can be reopened if picked up again.

**Pull request closed without merging**